kube/client.go

```diff
@@ -137,19 +137,21 @@ func (c *Client) Apply(path, namespace string, dryRun, prune, strict, kustomize
 		}
 	}
 
+	cmdStr := strings.Join(args, " ")
+
 	if strict {
-		tempKubeConfigFilepath, tempCertFilepath, err := c.CreateTempConfig(namespace, "kube-applier")
+		token, err := c.GetSAToken(namespace, "kube-applier")
 		if err != nil {
-			return "", "", fmt.Errorf("error creating temp config for kube-applier serviceaccount: %v", err)
+			return "", "", fmt.Errorf("error getting token for kube-applier serviceaccount: %v", err)
 		}
-		defer func() { os.Remove(tempKubeConfigFilepath); os.Remove(tempCertFilepath) }()
-		args = append(args, fmt.Sprintf("--kubeconfig=%s", tempKubeConfigFilepath))
+		args = append(args, fmt.Sprintf("--token=%s", token))
+		cmdStr += " --token=<omitted>"
 	} else if c.Server != "" {
 		args = append(args, fmt.Sprintf("--kubeconfig=%s", kubeconfigFilePath))
+		cmdStr += fmt.Sprintf("--kubeconfig=%s", kubeconfigFilePath)
```

Comment:
No whitespace?

Would it be tidier to keep a slice of strings until the end, and then turn it into a single string?

Reply:
Good spot on the whitespace. I am appending the final bit because I want to avoid adding the token as string in there. I will refactor

Reply:
example of what I am trying to achieve (from exp-1-gcp where this is running atm):
```
$ kubectl apply --server-dry-run=false -R -f /src/manifests/exp-1-gcp/sys-ingress-certs -l automaticDeployment!=off -n sys-ingress-certs --prune --prune-whitelist=apps/v1/DaemonSet --prune-whitelist=apps/v1/Deployment --prune-whitelist=apps/v1/StatefulSet --prune-whitelist=autoscaling/v1/HorizontalPodAutoscaler --prune-whitelist=batch/v1/Job --prune-whitelist=core/v1/ConfigMap --prune-whitelist=core/v1/Pod --prune-whitelist=core/v1/Service --prune-whitelist=core/v1/ServiceAccount --prune-whitelist=extensions/v1beta1/Ingress --prune-whitelist=networking.k8s.io/v1/NetworkPolicy --token=<omitted>
```

Reply:
I get it, but having to remember to format the output is error prone. Better keep it as close to the original before returning.